### PR TITLE
Update Fuseki Module documentation

### PR DESCRIPTION
**Pull request by afs**
See issue  https://github.com/apache/jena/issues/1897
and pull request https://github.com/apache/jena/pull/1898

### CODE REVIEW

File: source/documentation/fuseki2/fuseki-modules.md [hunk 1]
```diff
@@ -4,53 +4,80 @@ title: Fuseki Modules
 
 <em>Fuseki modules</em> are a mechanism to include extension code into a Fuseki
 server. Modules are invoked during the process of building a [Fuseki
-Main](./fuseki-main.html) server. The module can modify the server
+Main](./fuseki-main.html) server. A module can modify the server
 configuration, add new functionality, or react to a server being built and
 started.
 
-This feature was added in Jena version 4.3.0. They are an expeirmental feature
+This feature was added in Jena version 4.3.0. It is an expeirmental feature
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-modules.md:11]: s/expeirmental/experimental

File: source/documentation/fuseki2/fuseki-modules.md [hunk 2]
```diff
@@ -4,53 +4,80 @@ title: Fuseki Modules
...
-Fuseki Modules are loaded use the JDK
-[ServiceLoader](https://www.baeldung.com/java-spi) by being placing a jar file on the classpath,
+FusekiModules can be provided in two ways:
+
+* Loaded from additional jars on the classpath
+* Programmatically controlling the setup of the `FusekiServer` server.
+
+### Automatically loaded
+
+Fuseki Modules can be loaded use the JDK
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-modules.md:21]: s/use/using ?

File: source/documentation/fuseki2/fuseki-modules.md [hunk 3]
```diff
@@ -4,53 +4,80 @@ title: Fuseki Modules
...
-* `server` - access the built server
-* `serverBeforeStarting` - about to call "server.start()"
-* `serverAfterStarting` - just after calling "server.start()"
-* `serverStopped` - called as the server stop 
+* `prepare` - called at the start of the server
+   build steps before setting up the datasets.
+* `configured` - access and modify the setup. 
+   This is called after the server has been configured, before the server is built.
+   It defaults to calls to `configDataAccessPoint` for dataset being hosted by the server.
+* `server` - called after the built, beforee the return of `FusekiServerBuilder.build()`
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-modules.md:65]: s/beforee/before

File: source/documentation/fuseki2/fuseki-modules.md [hunk 4]
```diff
@@ -4,53 +4,80 @@ title: Fuseki Modules
...
+* `serverBeforeStarting` - called at the start of `server.start()`
+* `serverAfterStarting` - called at the end of `server.start()`
+* `serverStopped` - called as just after the server
+    has stopped in the `server.stop()` call.
   (note, this is not always called because a server can simply exit the JVM).
 
-A Fuseki module does not need to implement all these steps, the default for all
+A Fuseki module does not need to implement all these steps. The default for all
 of them is "do nothing". Usually, an extension will only be interested in
 certain stpes, like the configuration and registry information of
```
Comment (OyvindLGjesdal) [inline source/documentation/fuseki2/fuseki-modules.md:77]: stpes -> steps?

**Comment by afs** [comment]
Thank you both. Changes made.

File: source/documentation/fuseki2/fuseki-modules.md [hunk 5]
```diff
@@ -4,64 +4,107 @@ title: Fuseki Modules
...
-that will evolve based on experineace and use cases.
+This feature was added in Jena version 4.3.0. It is an experimental feature
+that will evolve based on feedback and use cases.
 
-Fuseki Modules are loaded use the JDK
-[ServiceLoader](https://www.baeldung.com/java-spi) by being placing a jar file on the classpath,
-together with any additional dependencies. The service loader looks for files
-`META-INF/services/org.apache.jena.fuseki.main.sys.FusekiModule` in the jar
+The interface for modules is `FusekiModule`; if automatcally loaded, the
+interface is `FusekiModule` which extends `FusekiModule`.
```
Comment (rvesse) [inline source/documentation/fuseki2/fuseki-modules.md:15]:
```suggestion
interface is `FusekiAutoModule` which extends `FusekiModule`.
```

File: source/documentation/fuseki2/fuseki-modules.md [hunk 6]
```diff
@@ -4,64 +4,107 @@ title: Fuseki Modules
...
+* Programmatically controlling the setup of the `FusekiServer` server.
+
+### Automatically loaded
+
+Fuseki Modules can be loaded using the JDK
+[ServiceLoader](https://www.baeldung.com/java-spi) by being placing
+a jar file on the classpath,
+together with any additional dependencies. These provide interface 
+`FusekiAutoModule`.
+The service loader is contreolled by file resources
```
Comment (rvesse) [inline source/documentation/fuseki2/fuseki-modules.md:29]:
```suggestion
The service loader is controlled by file resources
```

File: source/documentation/fuseki2/fuseki-modules.md [hunk 7]
```diff
@@ -4,64 +4,107 @@ title: Fuseki Modules
...
+be globally registered at this point (see the [Fuseki
+examples](https://github.com/apache/jena/tree/main/jena-fuseki2/jena-fuseki-main/src/test/java/org/apache/jena/fuseki/main/examples) directory).
+
+A `FusekiAutoModule` can provide a level, an integer, to control the order in which
+modules are invoked during server building. Lower numbers are invoked before larger
+numbers at each step.
+
+### Programmaticaly configuring a server
+
+If creating a Fuseki server from Java, the modules can be autoloaded as descriobed above,
```
Comment (rvesse) [inline source/documentation/fuseki2/fuseki-modules.md:50]:
```suggestion
If creating a Fuseki server from Java, the modules can be autoloaded as described above,
```

**Review by rvesse — APPROVED** [review]
Some minor typos

File: source/documentation/fuseki2/fuseki-modules.md (hunk 7 above)
Comment (afs) [reply source/documentation/fuseki2/fuseki-modules.md:50]: Done - thanks

**Pull request merged**